posthog/plugins/utils.py

```diff
@@ -11,6 +11,12 @@
 
 
 def parse_github_url(url: str, get_latest_if_none=False) -> Optional[Dict[str, Optional[str]]]:
+    private_token = None
+    if "?" in url:
+        url, query = url.split("?")
+        params = {k: v[0] for k, v in parse_qs(query).items()}
+        private_token = params.get("private_token", None)
+
```

Comment:
Since we're using this logic in a few places now, may be cleaner extracted to its own function.